shell/platform/embedder/BUILD.gn

```diff
@@ -122,11 +126,15 @@ template("embedder_source_set") {
         "embedder_external_texture_metal.mm",
         "embedder_surface_metal.h",
         "embedder_surface_metal.mm",
+        "embedder_surface_metal_impeller.h",
```

Comment:
For testing, (in a later patch is fine) I suggest converting the `impeller::EmbedderTest` fixture to be a parameterized fixture with the impeller parameterization setting up a context object with the right flag set. Then in the setup, you can skip the test on non-supported backends. The updates should be minimal. Likely just patching the TEST_F calls and instantiating the suite. But you'd double your test coverage.

Just a thought. And we can attempt this in a follow up patch.